fem/transfer.hpp

```diff
@@ -303,14 +433,25 @@ class L2ProjectionGridTransfer : public GridTransfer
       /// conservative left-inverse prolongation operation. This functionality
       /// is also provided as an Operator by L2Prolongation.
       void ProlongateTranspose(const Vector& x, Vector& y) const override;
+
+      void DeviceProlongateTranspose(const Vector& x, Vector& y) const;
+
+      /// Returns the inverse of an on-rank lumped mass matrix
+      void LumpedMassInverse(Vector& ML_inv) const;
+
       void SetRelTol(real_t p_rtol_) override;
       void SetAbsTol(real_t p_atol_) override;
+
    protected:
       /// Sets up the PCG solver (sets parameters, operator, and preconditioner)
       void SetupPCG();
-      /// Computes on-rank R and M_LH matrices.
+      void DeviceSetupPCG();
+
+      /// @brief Computes on-rank R and M_LH matrices. If true, computes mixed mass and/or
+      /// inverse lumped mass matrix error when compared to device implementation.
       std::pair<std::unique_ptr<SparseMatrix>,
-          std::unique_ptr<SparseMatrix>> ComputeSparseRAndM_LH();
+          std::unique_ptr<SparseMatrix>> ComputeSparseRAndM_LH(bool GetM_LHError=false,
+                                                               bool getML_invError=false);
```

Comment:
The comment "If true, ..." is not clear to me: if what is true?

The second parameter `getML_invError` is not used in the implementation and seems like it can be removed.

It looks like `GetM_LHError = true` triggers some verification for correctness -- if that is the case, a better name may be `check_LH_error` or something like that. This can also be clarified by documenting the argument in the doxygen comment.